macOS/DuckDuckGo/Sync/DeviceSyncCoordinator.swift

```diff
@@ -0,0 +1,164 @@
+//
+//  DeviceSyncCoordinator.swift
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import DDGSync
+import Combine
+import Common
+import SystemConfiguration
+import SyncUI_macOS
+import SwiftUI
+import Navigation
+import PixelKit
+import os.log
+import BrowserServicesKit
+
+protocol SyncDeviceFlowLaunching {
+    @MainActor
+    func startDeviceSyncFlow(completion: (() -> Void)?)
+}
+
+protocol DeviceSyncCoordinationDelegate: AnyObject {
+    @MainActor
+    func didEndFlow()
+}
+
+final class DeviceSyncCoordinator {
+    var cancellable: AnyCancellable?
+
+    @MainActor
+    init?(managementDialogModel: ManagementDialogModel = .init(), dialogController: SyncDialogController? = nil) {
+        self.managementDialogModel = managementDialogModel
+        guard let dialogController = dialogController ?? Self.createDialogController(managementDialogModel: managementDialogModel) else {
+            return nil
+        }
+        self.dialogController = dialogController
+        dialogController.coordinationDelegate = self
+    }
+
+    private let managementDialogModel: ManagementDialogModel
+    private let dialogController: SyncDialogController
+    private var syncWindowController: NSWindowController?
+
+    @MainActor
+    private func presentDialog(completion: (() -> Void)? = nil) {
+        guard !(syncWindowController?.window?.isVisible ?? false) else {
```

Comment:
The contents of this also used to be in `SyncPreferences.presentDialog`

Reply:
How about changing this guard to this, which I believe is simpler?
```suggestion
        guard syncWindowController?.window?.isVisible != true else {
```